Production_scripts/Process_extracts/extract_alarm_telecare.R

```diff
@@ -0,0 +1,228 @@
+#####################################################
+# Alarms Telecare Extract
+# Author: Zihao Li
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input - Social Care Care Home Episodes
+# Description - Match on Client data and Costs data
+#####################################################
+
+library(createslf)
+library(dplyr)
+library(lubridate)
+
+year <- check_year_format("1920")
+
+social_care_dir <-
+  fs::path("/conf/hscdiip/SLF_Extracts/Social_care")
+
+# Copy and paste from 03-Alarms_Telecare_data.R ----
+## Read Demographic file----
+
+sc_demographics <- haven::read_sav(fs::path(
+  social_care_dir,
+  paste0("sc_demographics_lookup_", latest_update()),
+  ext = "zsav"
+))
+
+## Query to database -------------------------------------------------------
+
+## set-up conection to platform
+db_connection <- phs_db_connection(dsn = "DVPROD")
+
+## read in data - social care 2 demographic
+at_full_data <- tbl(
+  db_connection,
+  dbplyr::in_schema("social_care_2", "equipment_snapshot")
+) %>%
+  select(
+    sending_location,
+    social_care_id,
+    period,
+    service_type,
+    service_start_date,
+    service_end_date
+  ) %>%
+  # fix bad period (2017, 2020 & 2021)
+  # TODO - ask SC team as last meeting they said to look at extract date - these dont relate.
+  # e.g. extract date later than period
+  mutate(
+    period = if_else(period == "2017", "2017Q4", period),
+    period = if_else(period == "2020", "2020Q4", period),
+    period = if_else(period == "2021", "2021Q4", period)
+  ) %>%
+  # order
+  arrange(sending_location, social_care_id) %>%
+  collect()
+
+
+## Data Cleaning-----------------------------------------------------
+
+# Work out the dates for each period
+# Record date is the last day of the quarter
+# qtr_start is the first day of the quarter
+pre_compute_record_dates <- at_full_data %>%
+  distinct(period) %>%
+  mutate(
+    record_date = yq(period) %m+% period(6, "months") %m-% days(1),
+    qtr_start = yq(period) %m+% period(3, "months")
+  )
+
+replaced_start_dates <- at_full_data %>%
+  # Replace missing start dates with the start of the FY
+  left_join(pre_compute_record_dates, by = "period") %>%
+  tidylog::mutate(
```

Comment:
We can drop the usage of tidylog
```suggestion
  tidylog::mutate(
```